enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,203 @@
+---
+title: automatic-detection-of-disks-in-Local-Storage-Operator
+authors:
+  - "@aranjan"
+reviewers:
+  - "@jrivera"
+  - "@jsafrane"
+  - "@hekumar"
+  - "@chuffman"
+  - "@rojoseph"
+  - "@sapillai"
+  - "@leseb"
+  - "@travisn"
+approvers:
+  - "@jrivera"
+  - "@jsafrane"
+  - "@hekumar"
+  - "@chuffman"
+creation-date: 2020-01-21
+last-updated: 2020-01-21
+status: implementable
+---
+
+# Automatic detection of Disks in Local-Storage-Operator 
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+When working with bare-metal OpenShift clusters, due to the absence of storage provisioners like that in the cloud, there is a lot of manual work to be done for consuming local storage from nodes. The idea is to automate the manual steps that are required for consuming local storage in an OpenShift cluster by extending the local storage operator (LSO).
+
+## Motivation
+
+To facilitate the consumption of locally-attached storage devices on OpenShift nodes in a cloud-native way.
+
+### Goals
+
+- Automatic detection of available disks from nodes which can be used as PVs for OpenShift-cluster.
+- Should respond to the attach/detach events of the disks/devices.
+- Should have options for filtering particular kind of disks based on properties such as name, size, manufacturer, etc.
+
+### Non-Goals
+
+## Proposal
+
+`local-storage-operator`is already capable of consuming local disks from the nodes and provisioning PVs out of it, but the disk/device paths needs to explicitly specified in the `LocalVolume` CR. The idea is to bring in one more layer of CR above it called `AutoDetectVolumes`.
+The controller for this CR will be responsible for automatically detecting disks/devices from the available nodes and creating/managing the life cycle of `LocalVolumes` which are governed by the `AutoDetectVolumes`.s
+
+### Risks and Mitigations
+
+- Detecting disks that contain data and/or are in-use.
+- Ensuring disks aren't re-detected as new or otherwise destroyed if their device path changes.
+
+## Design Details
+
+API scheme for `AutoDetectVolumes`:
+
+```
+type DeviceDiscoveryPolicyType string
+
+const (
+	// The DeviceDiscoveryPolicies that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	disk DeviceDiscoveryPolicyType = "disk"
+	part DeviceDiscoveryPolicyType = "part"
```

Comment:
These policies can be expanded in the future based on the requirement. I think initially we can start with the above two.